bobtimus/src/lib.rs

```diff
@@ -395,6 +420,141 @@ mod tests {
+
+        let input_alice = extract_input(
+            &client.get_raw_transaction(fund_alice_txid).await.unwrap(),
+            fund_address_alice,
+        )
+        .unwrap();
+
+        let (
+            final_address_alice,
+            _final_sk_alice,
+            _final_pk_alice,
+            _final_blinding_sk_alice,
+            _final_blinding_pk_alice,
+        ) = make_confidential_address();
+
+        // let address = client.getnewaddress().await.unwrap();
+        // let _txid = client
+        //     .send_asset_to_address(
+        //         &address,
+        //         Amount::from_btc(10.0).unwrap(),
+        //         Some(have_asset_id_bob),
+        //     )
+        //     .await
+        //     .unwrap();
+        // client.generatetoaddress(1, &mining_address).await.unwrap();
+
+        let mut bob = Bobtimus {
+            rng: &mut thread_rng(),
+            rate_service,
+            secp: Secp256k1::new(),
+            elementsd: client.clone(),
+            btc_asset_id: have_asset_id_bob,
+            usdt_asset_id: have_asset_id_alice,
+        };
+
+        let transaction = bob
+            .handle_create_buy_swap(CreateSwapPayload {
+                alice_inputs: vec![AliceInput {
+                    outpoint: input_alice.0,
+                    blinding_key: fund_blinding_sk_alice,
+                }],
+                address: final_address_alice,
+                btc_amount: sell_amount_bob,
+            })
+            .await
+            .unwrap();
+
+        let transaction = swap::alice_finalize_transaction(transaction, {
+            let commitment = input_alice.1.into_confidential().unwrap().value;
+            move |mut tx| async move {
+                let input_index = tx
+                    .input
+                    .iter()
+                    .position(|txin| fund_alice_txid == txin.previous_output.txid)
+                    .context("transaction does not contain input")?;
+                let mut cache = SigHashCache::new(&tx);
+
+                tx.input[input_index].witness.script_witness = sign_with_key(
+                    &SECP256K1,
+                    &mut cache,
+                    input_index,
+                    &fund_sk_alice,
+                    commitment.into(),
+                );
+
+                Ok(tx)
+            }
+        })
+        .await
+        .unwrap();
+
+        let _txid = client.send_raw_transaction(&transaction).await.unwrap();
+        let _txid = client.generatetoaddress(1, &mining_address).await.unwrap();
+
+        let utxos = client
+            .listunspent(
+                None,
+                None,
+                None,
+                None,
+                Some(ListUnspentOptions {
+                    asset: Some(have_asset_id_alice),
+                    ..Default::default()
+                }),
+            )
+            .await
+            .unwrap();
+
+        let error = 0.0001;
```

Comment:
What's the error for?

Reply:
It was to account for the fee. But good thing you asked, because Alice pays for the fee in this case so Bob actually gets the expected `redeem_amount_bob` exactly.

Furthermore, for the other test we can use `transaction.fee_in` to extract the exact fee paid in bitcoin and assert value equality, rather than using an error estimate.